diff --git a/src/nb_internals.h b/src/nb_internals.h
index 43881a4..e3c9326 100644
--- a/src/nb_internals.h
+++ b/src/nb_internals.h
@@ -937,7 +937,7 @@ NB_INLINE bool tstate_attached() noexcept {
 #if defined(Py_LIMITED_API) || defined(PYPY_VERSION)
     return false;
 #elif PY_VERSION_HEX < 0x030D0000
-    return _PyThreadState_UncheckedGet() != nullptr;
+    return PyGILState_Check() != 0;
 #else
     return PyThreadState_GetUnchecked() != nullptr;
 #endif
